stdlib/Dates/src/deprecated.jl

```diff
@@ -1,3 +1,67 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
+using Base: @deprecate, depwarn
+
 # 1.0 deprecations
+function (+)(x::AbstractArray{<:TimeType}, y::GeneralPeriod)
+    # depwarn("non-broadcasted arithmetic is deprecated for Dates.TimeType; use broadcasting instead", nothing)
```

Comment:
All the `nothing`s should be updated too?

Reply:
the name of the function that's being deprecated (in this case `:(+)`)

also mark all functions that contain a call to `depwarn` with `@noinline` so we can print the warning with correct location info